runtime/src/iree/base/internal/cpu.c

```diff
@@ -74,25 +56,33 @@ static void iree_cpu_initialize_from_platform(iree_allocator_t temp_allocator,
     }                                                             \
   } while (0)
 
-static void iree_cpu_initialize_from_platform(iree_allocator_t temp_allocator,
-                                              uint64_t* out_fields) {
-#if defined(IREE_ARCH_ARM_64)
+static void iree_cpu_initialize_from_platform_arm_64(uint64_t* out_fields) {
   IREE_QUERY_SYSCTL("hw.optional.arm.FEAT_DotProd", out_fields[0],
                     IREE_CPU_DATA_FIELD_0_AARCH64_HAVE_DOTPROD);
   IREE_QUERY_SYSCTL("hw.optional.arm.FEAT_I8MM", out_fields[0],
                     IREE_CPU_DATA_FIELD_0_AARCH64_HAVE_I8MM);
-#endif
 }
 
 #else
 
-static void iree_cpu_initialize_from_platform(iree_allocator_t temp_allocator,
-                                              uint64_t* out_fields) {
+static void iree_cpu_initialize_from_platform_arm_64(uint64_t* out_fields) {
   // No implementation available. CPU data will be all zeros.
 }
 
 #endif  // IREE_PLATFORM_*
 
+#endif  // defined(IREE_ARCH_ARM_64)
+
+static void iree_cpu_initialize_from_platform(iree_allocator_t temp_allocator,
+                                              uint64_t* out_fields) {
+#if defined(IREE_ARCH_ARM_64)
+  (void)temp_allocator;  // unused on ARM_64
```

Comment:
unless you need this for a compiler warning you can drop it - if it is required for a compiler warning then it should be outside the #if (as the #else would also need it) - we don't warn on unused arguments though so it shouldn't be required